au4/src/effects/effects_base/internal/effectsprovider.cpp

```diff
@@ -157,7 +162,12 @@ muse::Ret EffectsProvider::performEffect(AudacityProject& project, Effect* effec
             auto vr = valueRestorer(effect->mProgress, progress.get());
 
             assert(pInstanceEx); // null check above
-            returnVal = pInstanceEx->Process(settings);
+            try {
+                returnVal = pInstanceEx->Process(settings);
+            } catch (const ::AudacityException& e) {
+                // TODO: display message box using info from `e`.
+                returnVal = false;
```

Comment:
We need to add error codes (and text for them) to the effects module 
see https://github.com/musescore/MuseScore/wiki/Error-handling
or https://github.com/audacity/audacity/blob/master/au4/src/trackedit/trackediterrors.h 

and return `Ret` with the error code and text for the user, in the scenario show the error message.

Reply:
Will look into this - after lunch :D

Reply:
Read the doc, I know what you meant now.
But here it's important that we show an error dialog. (Could be, for example, that we try to generate 10s but the next clip is 5s away only ; since we don't support overlapping clips (yet), this branch would be entered. It's not unlikely at all.
I see
```cpp
void Au3BasicUI::DoShowErrorDialog(const BasicUI::WindowPlacement& placement, const TranslatableString& dlogTitle,
                                   const TranslatableString& message, const ManualPageID& helpPage,
                                   const BasicUI::ErrorDialogOptions& options)
{
    Q_UNUSED(placement);
    Q_UNUSED(dlogTitle);
    Q_UNUSED(message);
    Q_UNUSED(helpPage);
    Q_UNUSED(options);
}
```
Is there a plan already to get this implemented?

Reply:
Relevant background to the reporting of errors is Error identification in WCAG standards:
https://www.w3.org/WAI/WCAG22/Understanding/error-identification

This applies to Software as well as web pages.

Reply:
We do not use user interaction from logic (displaying messages, progress, etc.). The application is constantly improving, new ideas may come and, accordingly, logic functions, like bricks, can be used in completely different cases that do not imply those dialogs that should be shown now.
For example:
1. Some logic function may become part of a larger scenario, and the user needs to be informed not that this function failed, but that this larger scenario failed.
2. We can try one function, if it fails, try another one... without telling the user.
3. For some reason, we can execute some logic function many times in a row, but we don’t want to show the user a dozen dialogs...
4. Or we want to avoid what has already happened in effects - when executing an effect, two dialog progresses can now be shown, the second one on top of the first one.
and etc.
For this reasons, the usual approach is to return an error code (possibly with text), for this we use the Ret type. And somewhere above, at the level of the view model, controller or scenario, formed and display a message to the user. (edited) 

But if we can't avoid this in the AU3 code now, we'll have to implement this function and live with it for some time.

> Could be, for example, that we try to generate 10s but the next clip is 5s away only 
This is the task of forming information about the error, and the one who calls it can show it... if necessary.